athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/TypeRowWriter.java

```diff
@@ -0,0 +1,104 @@
+import org.slf4j.LoggerFactory;
+import com.amazonaws.athena.connector.lambda.data.writers.holders.NullableVarCharHolder;
+import org.apache.arrow.vector.holders.NullableIntHolder;
+import org.apache.arrow.vector.holders.NullableFloat4Holder;
+import org.apache.arrow.vector.holders.NullableFloat8Holder;
+
+public class TypeRowWriter {
+
+    public static RowWriterBuilder writeRowTemplate(RowWriterBuilder rowWriterBuilder, Field field) {
+
+        Logger logger = LoggerFactory.getLogger(TypeRowWriter.class);
+
+        ArrowType arrowType = field.getType();
+        Types.MinorType minorType = Types.getMinorTypeForArrowType(arrowType);
+
+        switch (minorType) {
+
+            case FLOAT8:
+                rowWriterBuilder.withExtractor(field.getName(),
+                        (Float8Extractor) (Object context, NullableFloat8Holder value) -> {
+                            value.isSet = 1;
+
+                            Map<Object, Object> obj = (Map<Object, Object>) context;
+                            ArrayList<Object> objValues = (ArrayList)obj.get(field.getName());
+
+                            value.value = Float.parseFloat(objValues.get(0).toString());
+                        });
+
+                break;
+
+            case VARCHAR:
+
+                rowWriterBuilder.withExtractor(field.getName(),
+                        (VarCharExtractor) (Object context, NullableVarCharHolder value) -> {
+                            value.isSet = 1;
+                            Map<Object, Object> obj = (Map<Object, Object>) context;
+                            ArrayList<Object> objValues = (ArrayList)obj.get(field.getName());
+
+                            value.value = objValues.get(0).toString();
+                        });
+                break;
+
+            case FLOAT4:
+
+                rowWriterBuilder.withExtractor(field.getName(),
+                        (Float4Extractor) (Object context, NullableFloat4Holder value) -> {
+                            value.isSet = 1;
+
+                            Map<Object, Object> obj = (Map<Object, Object>) context;
+                            ArrayList<Object> objValues = (ArrayList)obj.get(field.getName());
+
+                            value.value = Float.parseFloat(objValues.get(0).toString());
+                        });
+                break;
+
+            case INT:
+                rowWriterBuilder.withExtractor(field.getName(),
+                        (IntExtractor) (Object context, NullableIntHolder value) -> {
+                            value.isSet = 1;
+
+                            Map<Object, Object> obj = (Map<Object, Object>) context;
+                            ArrayList<Object> objValues = (ArrayList)obj.get(field.getName());
+                            value.value = Integer.parseInt(objValues.get(0).toString());
+                        });
+                break;
+
+            default:
+                break;
+        }
+
+        return rowWriterBuilder;
```

Comment:
no need to return. When you call rowWriterBuilder.withExtrator(...) you are mutating the internal state of rowWriterBuilder and since java is pass by reference by default there is no need for the return.

Reply:
re-factored